cmd/stellar-rpc/internal/config/options.go

```diff
@@ -84,7 +85,25 @@ func (cfg *Config) options() Options {
 			Name:         "stellar-captive-core-http-port",
 			Usage:        "HTTP port for Captive Core to listen on (0 disables the HTTP server)",
 			ConfigKey:    &cfg.CaptiveCoreHTTPPort,
-			DefaultValue: uint(11626),
+			DefaultValue: uint16(defaultCaptiveCoreHTTPPort),
+		},
+		{
+			Name:         "stellar-captive-core-http-query-port",
+			Usage:        "HTTP port for Captive Core to listen on for high-performance queries like /getledgerentry (0 disables the HTTP server, must not conflict with CAPTIVE_CORE_HTTP_PORT)",
+			ConfigKey:    &cfg.CaptiveCoreHTTPQueryPort,
+			DefaultValue: uint16(0), // Disabled by default, although it normally uses 11628
+		},
+		{
+			Name:         "stellar-captive-core-http-query-thread-pool-size",
+			Usage:        "Number of threads to use by Captive Core's high-performance query server",
+			ConfigKey:    &cfg.CaptiveCoreHTTPQueryThreadPoolSize,
+			DefaultValue: uint16(runtime.NumCPU()), //nolint:gosec
```

Comment:
Is this a safe default? We want to leave CPUs for RPC itself.. possible suggestion,

```suggestion
			DefaultValue: uint16(min(1, runtime.NumCPU() - 2)), //nolint:gosec
```

Reply:
Uhm. What if there is a single or 2 CPUs. As far as I understand, the query server is IO bound, so the threads will be blocked waiting for disk most of the time.

Reply:
Sorry should've been `max` not `min` in the diff :facepalm: - so at least 2, but otherwise `CpuCount - 2`. 

Good point re: disk-bound, but not true for live state (live state will be wholly in memory). So I do think we should certainly leave a CPU completely free for RPC itself. Maybe for the rest of Captive Core, as well? A big `getLedgerEntries` stalling ingestion by hogging CPU would not be good.

@SirTyson opinions here?

Reply:
I highly suspect this endpoint to be IO bound, even with in-memory state. Tbh I haven't tested with a ton of cores yet, that sort of config setting is probably something you all are better equipped to do, but I'd assume these are IO threads and not compute threads.